File: 2023/src/day6/two.rs
pub(crate) fn solve(lines: Vec<String>) -> String {
    let time = line_to_num(lines.get(0).unwrap());
    let distance = line_to_num(lines.get(1).unwrap());

    (0..time)
        .filter(|i| (time - i) * i > distance)
        .count()
        .to_string()
}

fn line_to_num(line: &str) -> i64 {
    line.split(":")
        .nth(1)
        .unwrap()
        .split(" ")
        .collect::<Vec<&str>>()
        .join("")
        .parse::<i64>()
        .unwrap()
}

#[cfg(test)]
mod tests {
    use crate::{day6::two::solve, utils::util};

    #[test]
    fn test_sample() {
        let lines = util::lines_in("../../aoc-files/2023/day6/input");
        assert_eq!("71503", solve(lines))
    }

    #[test]
    fn test_input() {
        let lines = util::lines_in("../../aoc-files/2023/day6/input1");
        assert_eq!("32607562", solve(lines))
    }
}

File: 2020/03/two.ts
import { promises as fs } from 'fs';

async function solve() {
    const input = await fs.readFile("../../../aoc-files/2020/03/input");
    const board = input.toString().split("\n").map(line => line.split(""));
    let trees = numberOfTrees(board, 1, 1);
    trees *= numberOfTrees(board, 3, 1);
    trees *= numberOfTrees(board, 5, 1);
    trees *= numberOfTrees(board, 7, 1);
    trees *= numberOf2Trees(board, 1, 2);

    console.log(trees);
}

function numberOfTrees(board: string[][], right: number, down: number): number {
    const [m, n] = [board.length, board[0].length];
    let i = 0;
    let j = 0;
    let trees = 0;
    console.log(m, n);

    while (i < m - 1) {
        j += right;
        j = j % n;
        i += down;
        if (board[i][j] === '#') trees++;
    }
    console.log(trees);
    return trees;
}

function numberOf2Trees(board: string[][], right: number, down: number): number {
    const [m, n] = [board.length, board[0].length];
    let i = 0;
    let j = 0;
    let trees = 0;
    console.log(m, n);

    while (i < m - 2) {
        j += right;
        j = j % n;
        i += down;
        if (board[i][j] === '#') trees++;
    }
    console.log(trees);
    return trees;
}

console.log("Solving");
solve();